tests/core/pyspec/eth2spec/test/helpers/consolidations.py

```diff
@@ -0,0 +1,18 @@
+from eth2spec.test.helpers.withdrawals import (
+    set_eth1_withdrawal_credential_with_balance
+)
+
+
+def prepare_switch_to_compounding_request(spec, state, validator_index, address=None):
+    if address is not None:
+        set_eth1_withdrawal_credential_with_balance(spec, state, validator_index, address=address)
+
+    validator = state.validators[validator_index]
+    if not spec.has_execution_withdrawal_credential(validator):
+        set_eth1_withdrawal_credential_with_balance(spec, state, validator_index)
```

Comment:
This chunk of code seems strange. Why are there two conditions? Could it be replaced with:

```suggestion
    validator = state.validators[validator_index]
    if not spec.has_execution_withdrawal_credential(validator):
        set_eth1_withdrawal_credential_with_balance(spec, state, validator_index, address=address)
```

If an address isn't provided, it would behave the same.

https://github.com/ethereum/consensus-specs/blob/60cd223fb5064078ccdb65a14a969c685c6f12b6/tests/core/pyspec/eth2spec/test/helpers/withdrawals.py#L31

The first condition is a bit worrisome, as it will overwrite a compounding withdrawal cred.

Same concern applies to `prepare_withdrawal_request` below.

Reply:
The idea behind the logic here:
* set the address if it is supplied
* then if the validator has neither `0x01` nor `0x02` withdrawal creds type, set some default eth1 creds